src/components/interface/capmgr/capmgr.h

```diff
@@ -0,0 +1,17 @@
+#ifndef CAPMGR_H
+#define CAPMGR_H
+
+#include <cos_kernel_api.h>
+#include <cos_defkernel_api.h>
+
+thdcap_t  capmgr_initthd_create(spdid_t child, thdid_t *tid);
+thdcap_t  capmgr_initaep_create(spdid_t child, struct cos_aep_info *aep, int owntc, asndcap_t *sndret);
+thdcap_t  capmgr_thd_create(cos_thd_fn_t fn, void *data, thdid_t *tid);
+thdcap_t  capmgr_aep_create(struct cos_aep_info *a, cos_aepthd_fn_t fn, void *data, int owntc);
+thdcap_t  capmgr_ext_thd_create(spdid_t child, int idx, thdid_t *tid);
```

Comment:
I think that we need to add a file in `docs/` for `naming_conventions.md` including things like:

- `_intern` - an internal API to a component (even if the function can't be `static`)
- `_ext` - and extended API that includes all options (in place of method overloading in languages like C++)
- `capin` - a capability offset into another capability table (often the previous argument)
- `ct` - capability table
- `Xcap` - where `X` is `ct`, `pt`, `pgtbl`, `thd`, etc...; a capability to that resources.
- `capfrom` and `capto` - the capability offset that is being operated on, and the capability offset where that operation is resulting (e.g. copying from an offset, to another one.  Similarly, `ctfrom` and `ctto` 

....

Reply:
kk. understood. I'll follow that.